octopusdeploy/schema_project_scheduled_trigger.go

```diff
@@ -0,0 +1,543 @@
+			{
+				"start_time":   onceDailyScheduleFilter.Start.Format(filters.RFC3339NanoNoZone),
+				"days_of_week": onceDailyScheduleFilter.Days,
+			},
+		}
+		flattenedProjectScheduledTrigger["timezone"] = onceDailyScheduleFilter.TimeZone
+	} else if filterType == filters.ContinuousDailySchedule {
+		continuousDailyScheduleFilter := projectScheduledTrigger.Filter.(*filters.ContinuousDailyScheduledTriggerFilter)
+
+		flattenedProjectScheduledTrigger["continuous_daily_schedule"] = []map[string]interface{}{
+			{
+				"interval":        continuousDailyScheduleFilter.Interval.String(),
+				"hour_interval":   continuousDailyScheduleFilter.HourInterval,
+				"minute_interval": continuousDailyScheduleFilter.MinuteInterval,
+				"run_after":       continuousDailyScheduleFilter.RunAfter.Format(filters.RFC3339NanoNoZone),
+				"run_until":       continuousDailyScheduleFilter.RunUntil.Format(filters.RFC3339NanoNoZone),
+				"days_of_week":    continuousDailyScheduleFilter.Days,
+			},
+		}
+		flattenedProjectScheduledTrigger["timezone"] = continuousDailyScheduleFilter.TimeZone
+	} else if filterType == filters.DaysPerMonthSchedule {
+		daysPerMonthScheduleFilter := projectScheduledTrigger.Filter.(*filters.DaysPerMonthScheduledTriggerFilter)
+		flattenedProjectScheduledTrigger["days_per_month_schedule"] = []map[string]interface{}{
+			{
+				"start_time":            daysPerMonthScheduleFilter.Start.Format(filters.RFC3339NanoNoZone),
+				"monthly_schedule_type": daysPerMonthScheduleFilter.MonthlySchedule.String(),
+				"date_of_month":         daysPerMonthScheduleFilter.DateOfMonth,
+				"day_number_of_month":   daysPerMonthScheduleFilter.DayNumberOfMonth,
+				"day_of_week":           filters.Weekday.String(*daysPerMonthScheduleFilter.Day),
+			},
+		}
+		flattenedProjectScheduledTrigger["timezone"] = daysPerMonthScheduleFilter.TimeZone
+	} else if filterType == filters.CronExpressionSchedule {
+		cronExpressionScheduleFilter := projectScheduledTrigger.Filter.(*filters.CronScheduledTriggerFilter)
+		flattenedProjectScheduledTrigger["cron_expression_schedule"] = []map[string]interface{}{
+			{
+				"cron_expression": cronExpressionScheduleFilter.CronExpression,
+			},
+		}
+		flattenedProjectScheduledTrigger["timezone"] = cronExpressionScheduleFilter.TimeZone
+	}
+
+	return flattenedProjectScheduledTrigger
+}
+
+func expandProjectScheduledTrigger(projectScheduledTrigger *schema.ResourceData, client *client.Client) (*triggers.ProjectTrigger, error) {
+	name := projectScheduledTrigger.Get("name").(string)
+	description := projectScheduledTrigger.Get("description").(string)
+	isDisabled := projectScheduledTrigger.Get("is_disabled").(bool)
+	timezone := projectScheduledTrigger.Get("timezone").(string)
+
+	projectId := projectScheduledTrigger.Get("project_id").(string)
+	spaceId := projectScheduledTrigger.Get("space_id").(string)
+	project, err := projects.GetByID(client, spaceId, projectId)
```

Comment:
Can the project resource be passed into this function instead of a `client`? That way nothing in this file needs to be concerned about the client.